auter

```diff
@@ -301,6 +323,21 @@ if [[ -z "${CONFIGSET}" ]]; then
   quit 5
 fi
 
+if [[ "${ONLYINSTALLFROMPREP}" == "yes" ]]; then
+  if [[ ! -d "${DOWNLOADDIR}/${CONFIGSET}" ]]; then
+    install -m 755 -o root -g root -d ${DOWNLOADDIR}/${CONFIGSET}
+  elif [[ $(stat -c %G%U%a "${DOWNLOADDIR}") != rootroot[0-9][0-9][0,1,4,5] ]]; then
+    logit "ERROR: ${DOWNLOADDIR}/${CONFIGSET} does not have the correct permissions."
+    quit 3
+  fi
+  DOWNLOADOPTION="--downloaddir=${DOWNLOADDIR}/${CONFIGSET}"
```

Comment:
DOWNLOADOPTION has been moved to prepare_updates function.
UPDATEACTION check has now been moved to the apply_updates function.

We should be using built-in functionality of yum. The fact that dnf does not yet have this feature should not make us disregard built in functions of yum.